src/ts/components/settingspanelitem.ts

```diff
@@ -11,47 +11,74 @@ import {PlaybackSpeedSelectBox} from './playbackspeedselectbox';
 import { PlayerAPI } from 'bitmovin-player';
 import { LocalizableText } from '../localization/i18n';
 
+/**
+ * Configuration interface for a {@link SettingsPanelItem}
+ *
+ * @category Configs
+ */
+export interface SettingsPanelItemConfig extends ContainerConfig {
+  /**
+   * The label component or the text for the label.
+   */
+  label?: LocalizableText | Component<ComponentConfig>;
+  /**
+   * The component that configures a setting.
+   */
+  setting?: Component<ComponentConfig>;
+  /**
+   * If the setting should be added as a component to this item.
+   */
+  addSettingAsComponent?: boolean;
+}
+
 /**
  * An item for a {@link SettingsPanelPage},
  * Containing an optional {@link Label} and a component that configures a setting.
  * If the components is a {@link SelectBox} it will handle the logic of displaying it or not
  *
  * @category Components
  */
-export class SettingsPanelItem extends Container<ContainerConfig> {
+export class SettingsPanelItem<Config extends SettingsPanelItemConfig> extends Container<Config> {
 
   private label: Component<ComponentConfig>;
-  protected setting: Component<ComponentConfig>;
+  protected setting: Component<ComponentConfig> | null;
 
   private settingsPanelItemEvents = {
-    onActiveChanged: new EventDispatcher<SettingsPanelItem, NoArgs>(),
+    onActiveChanged: new EventDispatcher<SettingsPanelItem<Config>, NoArgs>(),
   };
 
-  constructor(label: LocalizableText | Component<ComponentConfig>, setting: Component<ComponentConfig>, config: ContainerConfig = {}, addSettingAsComponent: boolean = true) {
```

Comment:
I refactored the constructor to follow our config-based approach, which we follow for almost all other Components.